content/blog/2022-04-26-Superoptimizer.md

```diff
@@ -0,0 +1,83 @@
++++
+title = "Superoptimizer: a look at the smallest program"
+[[extra.authors]]
+name = "Jonathan Tran"
+[[extra.authors]]
+name = "Victor Giannakouris"
++++
+
+## Introduction
+
+The majority of compilers utilize optimization to improve code, in terms of runtime performance, code size, and other factors. However, optimization does not necessarily generate the optimal program; the optimization process merely generates a better candidate program. In the Superoptimization paper by Massalin,, a different paradigm is introduced: optimization is performed to generate the minimal sized program, producing programs that are smaller compared to traditionally compiler-optimizer programs, as well as human hand-optimized programs [^1].
+
+As an illustration of Superoptimizer’s capabilities, consider an example centered on compiling the signum function, which is defined as follows: 
+
+```int signum(x) {
+	If (x > 0) return 1;
+	Else if (x <  0) return -1;
+	Else return 0;
```

Comment:
I know this is pseudocode, but I can't help but notice it would be valid C if the keywords were not capitalized (If -> if, etc.)…

Reply:
I've changed this to be C code in the markdown file.